libraries/RC_Channel/RC_Channels.cpp

```diff
@@ -234,6 +240,52 @@ uint32_t RC_Channels::enabled_protocols() const
     return uint32_t(_protocols.get());
 }
 
+
+/*
+  check for pilot input on rudder stick for arming/disarming
+*/
+void RC_Channels::rudder_arm_disarm_check()
+{
+    const RC_Channel *channel = rc().get_arming_channel();
+    if (channel == nullptr) {
+        return;
+    }
+
+    if (!has_valid_input() ||
+        abs(channel->get_control_in()) <= 4000) {
+        // not trying to (or no longer trying to) arm or disarm - or
+        // we don't have valid RC input.
+        rudder_arm_timer = 0;
+        rudder_10_second_timer = 0;
+        return;
+    }
+
+    const uint32_t now = AP_HAL::millis();
+    if (rudder_arm_timer == 0) {
+        // first time we've seen the attempt
+        rudder_arm_timer = now;
+    } else if (now - rudder_arm_timer >= 3000) {
+        rudder_arm_timer = 0;  // do not reattempt for a while
+        // time to try to arm or disarm:
+        if (channel->get_control_in() > 4000) {
```

Comment:
It's fair to not do it now, but for future reference we shouldn't be hardcoding 4000 here. We should be fetching the channel's range, then comparing to a given percentage of that range. (That way we have an option of normalizing to something logical like a float or higher resolution in the future)